### ISSUE TITLE
Decrease latency of unilm Question Generation task- optimisations

### ISSUE THREAD

**Issue by pdcoded** [microsoft/unilm#124]
I am using the finetuned checkpoint of unilm-v1 for squad question generation task that's provided here https://github.com/microsoft/unilm/tree/master/unilm-v1

Since the model, it's trained on is - unilmv1-large-cased, it is very slow during decoding when I use it as a microservice in an Nvidia-p4 k8s cluster (almost 30-40 secs for 1 req with stepped memory usage increase). Although the accuracy is really good since it implicitly does decent coreference resolution through attention while asking questions.

In order of priority, with the tradeoff of accuracy, what would be the best step for decreasing latency::

1. Training minilm and decoding it - Can we get fine-tuned checkpoint for this like unilm-v1 

2. Any suggestions for sticking to same accuracy as unilm but reduced inference time - like tensorrt optimizations?

**Comment by pdcoded** [comment]
FYI I ran experiment - FineTuning minilm (minilm-l12-h384-uncased) using s2s-ft Model on squad data. Results are impressive::
As stated before, on a p4 GPU instance, unilm was taking close to 40 secs.

Now - with this smaller pruned model, on n1-small CPU instance, able to run in sub 4secs.

And the results are comparable to unilm too on my dataset.

Closing the issue - happy to provide more info if someone reopens. Cheers !!

**Comment by donglixp** [comment]
Awesome! Thanks for the feedback!

**Comment by ArijRB** [comment]
@pdcoded could you please give QG usage example for multilingual MiniLM?
Thanks in advance

**Comment by WenhuiWang0824** [comment]
Hi @ArijRB,

Please refer to the comment in [https://github.com/microsoft/unilm/issues/103].

Thanks
